### PR TITLE
[GEN-1622] remove sample class filter

### PR DESCRIPTION
**Pull request by danlu1**
Problem:

cfDNA samples are not released publicly.

Solution:
Make cfDNA samples public in main GENIE releases by removing sample_class_filter.

Testing:
Unit test has been added. And integration test has been done. Comparison report between `syn11611431.111` (with sample_class_filter) and `syn11611431.117` (without sample_class_filter) has been generated and got the expected result: two cfDNA sample rows are propagated.

Screenshot 2024-11-25 at 9 53 21 PM

### CODE REVIEW

File: genie/consortium_to_public.py [hunk 1]
```diff
@@ -147,7 +144,19 @@ def consortiumToPublic(
         query_string=f"SELECT * FROM {clinical_tier_release_scope_synid} where releaseScope = 'public'",
     )
 
+    # check if SAMPLE_CLASS is present
+    if not process_functions.check_values_in_column(
```
Comment (thomasyu888) [inline genie/consortium_to_public.py:152]: These are great post processing checkers, eventually these will most likely be superceded with the new data quality framework rixing may want to put together.

**Review by thomasyu888 — COMMENTED** [review]
🔥 LGTM! will leave for rixing for final review.

File: genie/consortium_to_public.py [hunk 2]
```diff
@@ -147,7 +144,19 @@ def consortiumToPublic(
...
+        publicRelease, "fieldName", "SAMPLE_CLASS"
+    ):
+        logger.error("Must have SAMPLE_CLASS column in the public release scope.")
```
Comment (rxu17) [inline genie/consortium_to_public.py:155]: Did this logging show up on the console log for a nextflow tower run for you? I think if not, we may have to do [something like this](https://github.com/Sage-Bionetworks/Genie/blob/develop/genie/extract.py#L16-L19) for everywhere we get `logging` to get it to show up in the console.

File: tests/test_process_functions.py [hunk 3]
```diff
@@ -715,3 +714,74 @@ def test_that_create_missing_columns_returns_expected_output_with_multi_col_df()
...
+)
+def test_check_values_in_column_no_column(input_df, col, values):
+    with patch.object(process_functions, "logger") as mock_logger:
+        results = process_functions.check_values_in_column(input_df, col, values)
+    mock_logger.error.assert_called_once_with(
+        "Must have test_col column in the dataframe."
+    )
+
+
+@pytest.mark.parametrize(
```
Comment (rxu17) [inline tests/test_process_functions.py:732]: Could you add a test case for an empty dataframe? One that has the columns req in the check but no rows of data

**Review by rxu17 — APPROVED** [review]
LGTM! Just a couple of comments

File: genie/consortium_to_public.py (hunk 2 above)
Comment (danlu1) [reply genie/consortium_to_public.py:155]: The logging shows in the console when I run locally but I did't check it in tower. Let me check it.

Comment (danlu1) [reply genie/consortium_to_public.py:155]: You are right. Thanks for catching this. The logging is not showing in the tower console or log file. I tweaked the code by adding stdout_handler.

File: tests/test_process_functions.py (hunk 3 above)
Comment (danlu1) [reply tests/test_process_functions.py:732]: Done. It throws False as expected.

File: genie/consortium_to_public.py (hunk 2 above)
Comment (danlu1) [reply genie/consortium_to_public.py:155]: The logging info shows up in the Tower console now. 
Screenshot 2024-12-03 at 11 27 22 AM
